### PR TITLE
Fix some backward compatibility issue

### PR DESCRIPTION
**Pull request by quink-black**

### CODE REVIEW

**Comment by kylophone** [comment]
Can you please give some context for these changes?

**Comment by quink-black** [comment]
> Can you please give some context for these changes?

It's in the commit message:

1. model: fix misleading log

    Deprecated means it works but will be removed. The support of pkl
    model has been removed. The log message and log level make user
    wondering why it doesn't work.

2.     compute_vmaf: use default format when log_path is specified but log_fmt is missing

    For backward compatibility. libvmaf has default log_fmt is not specified in previous version.

3.    compute_vmaf: add sub to log_fmt_map

Support a new format through old API.

**Comment by quink-black** [comment]
By the way, these issues were found when using libvmaf via FFmpeg. I'm wondering if there is some progress in

https://github.com/Netflix/vmaf/issues/753

File: libvmaf/src/model.c [hunk 1]
```diff
@@ -137,12 +137,13 @@ int vmaf_model_load_from_path(VmafModel **model, VmafModelConfig *cfg,
...
-        vmaf_log(VMAF_LOG_LEVEL_WARNING,
+        vmaf_log(VMAF_LOG_LEVEL_ERROR,
                  "could not read model from path: \"%s\"\n", path);
         char *ext = strrchr(path, '.');
         if (ext && !strcmp(ext, ".pkl")) {
-            vmaf_log(VMAF_LOG_LEVEL_WARNING,
-                     "pkl model files have been deprecated, use json\n");
+            vmaf_log(
+                VMAF_LOG_LEVEL_ERROR,
+                "support for pkl model files have been removed, use json\n");
```
Comment (slhck) [inline libvmaf/src/model.c:146]: Minor nit: "support … **has** been removed, …"

File: libvmaf/src/model.c [hunk 2]
```diff
@@ -282,12 +283,13 @@ int vmaf_model_collection_load_from_path(VmafModel **model,
...
-        vmaf_log(VMAF_LOG_LEVEL_WARNING,
+        vmaf_log(VMAF_LOG_LEVEL_ERROR,
                  "could not read model collection from path: \"%s\"\n", path);
         char *ext = strrchr(path, '.');
         if (ext && !strcmp(ext, ".pkl")) {
-            vmaf_log(VMAF_LOG_LEVEL_WARNING,
-                     "pkl model files have been deprecated, use json\n");
+            vmaf_log(
+                VMAF_LOG_LEVEL_ERROR,
+                "support for pkl model files have been removed, use json\n");
```
Comment (slhck) [inline libvmaf/src/model.c:292]: Same comment as above; use "has" instead of "have"

**Comment by slhck** [comment]
This is indeed something rather confusing, also seeing that some precompiled ffmpeg binaries eventually switched back to libvmaf 1.x because end users had issues with the 2.x version and the model files. (I don't know what these issues were. I'm just paraphrasing a maintainer.)

**Pull request merged**